### PR TITLE
Free memory in Parodus during exit/shutdown

### PR DESCRIPTION
**Pull request by shilpa24balaji**
To free memory of few config in Parodus during the shutdown case

### CODE REVIEW

**Comment by shilpa24balaji** [comment]
@sadhyama Not able to resolve the security warning even after pointers are set to null

File: src/config.c [hunk 1]
```diff
@@ -711,6 +711,50 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
     return 0;
 }
 
+void free_cfg(ParodusCfg *cfg)
+{
+    if (cfg->mtls_client_cert_path != NULL )
```
Comment (sadhyama) [inline src/config.c:716]: Can you add cfg NULL check and assign it to NULL at the end once elements are freed .

Comment (shilpa24balaji) [reply src/config.c:716]: cfg I traced back is pointer to static global struct variable so ideally we should not assign it NULL 
https://github.com/xmidt-org/parodus/blob/master/src/config.c#L44

Comment (shilpa24balaji) [reply src/config.c]: Added NULL check but not able to run guardrails again to confirm

**Review by schmidtw — APPROVED** [review]

**Comment by schmidtw** [comment]
This code looks correct to me.  I'm merging it.

File: src/conn_interface.c [hunk 2]
```diff
@@ -231,6 +231,9 @@ void createSocketConnection(void (* initKeypress)())
     nopoll_ctx_unref(ctx);
     nopoll_cleanup_library();
     curl_global_cleanup();
+    clear_metadata();
+    rdk_logger_deinit();
```
Comment (suvi-123) [inline src/conn_interface.c:235]: Facing build error due to this in local Ubuntu pc,

/usr/bin/ld: CMakeFiles/parodus.dir/conn_interface.c.o: in function `createSocketConnection':
conn_interface.c:(.text+0x89b): undefined reference to `rdk_logger_deinit'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/parodus.dir/build.make:519: src/parodus] Error 1
make[1]: *** [CMakeFiles/Makefile2:1274: src/CMakeFiles/parodus.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

Comment (sadhyama) [reply src/conn_interface.c:235]: @shilpa24balaji Could you please help to check this ? 
rdk_logger_deinit func is not available in local pc code hence this error.
For now we are proceeding by disabling this func call from conn_interface.c

**Pull request merged**